src/util/spreadsheet/misc.py

```diff
@@ -1,10 +1,13 @@
-from .. import json
+from .. import json, pg
+
+BATCH_SIZE = 10
 
 
 def iter_commands(cr, like_all=(), like_any=()):
     if not (bool(like_all) ^ bool(like_any)):
         raise ValueError("Please specify `like_all` or `like_any`, not both")
-    cr.execute(
+    ncr = pg.named_cursor(cr, itersize=BATCH_SIZE)
```

Comment:
Using a context manager you do not need to close it explicitely[^1].

```suggestion
    with pg.named_cursor(cr, itersize=BATCH_SIZE) as ncr:
```

That said, this is just in the name of a more pythonic implementation. IOW: imo you can keep your current version, if you like it better.

[^1]: https://github.com/odoo/odoo//blob/bdc28e53b7c426640e30b900064442f19c946d9e/odoo/sql_db.py#L221-L238